### PR TITLE
Fix alluxio tarball owner in docker

### PR DESCRIPTION
**Pull request by ssz1997**
The owner of files in `opt/alluxio` is the owner of the tarball. If the user builds the tarball locally as non-alluxio user, the user alluxio (uid 1000) inside the docker container may not have the permission to write into `alluxio-env.sh`, which is a required step in `entrypoint.sh`, and thus the cluster is not able to start.

solves #15309

### CODE REVIEW

**Comment by ssz1997** [comment]
We can't simply do a chown at the "ADD" time, because locally added tarball will be automatically extracted in docker, and the files inside the tarball after extraction won't be "chowned".

Another way is to first do "ADD" then do chown in another step. However the "chown" step will create another layer and thus increasing the overall image size.

So first do "ADD", then "COPY" with chown would be the best solution.

File: integration/docker/Dockerfile [hunk 1]
```diff
@@ -36,6 +35,9 @@ RUN if [ ${ENABLE_DYNAMIC_USER} = "true" ] ; then \
        chmod -R 777 /opt/* ; \
     fi
 
+# For production use, we use alpine as base image to minimize alluxio image size.
+FROM alpine:3.10.2 AS final
```
Comment (LuQQiu) [inline integration/docker/Dockerfile:39]: Why we 
```
FROM alpine:3.10.2 AS alluxio-extractor
FROM alpine:3.10.2 AS final
```
?

**Review by LuQQiu — APPROVED** [review]
LGTM except one minor comment

**Review by LuQQiu — APPROVED** [review]

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Comment by LuQQiu** [comment]
@Xenorith Please help cherry-pick to 2.8

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.8 please

**Comment by LuQQiu** [comment]
@Xenorith can you help retry the cherr-pick, if fail again, we go for manual cherry-pick

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.8 please

**Pull request merged**